Pages Added

Viraj-Sonavane opened the pull request:
1. Login and Registration page added.
2. Separate Header for Login and Registration Page added.
3. Top header for Home page added.

Viraj-Sonavane committed:
Page addition

1.Login Page and  Register Page added
2.Different Header for both Login and Registration Page added.
3. Top added for Home Page

Viraj-Sonavane committed:
Page additions

1.Login Page and Register Page added
2.Different Header for both Login and
   Registration Page added.
3.Top Header added for Home Page

RetroJect reviewed (changes requested):
Let's remove all of the images we're not using and move some of the files into the structure we already have. We don't need the MainLayout file either as that's taken care of by our top-level App component. Authentication/Header can replace our Header component and be moved to take its place.

I have other suggestions but I'll wait until these are addressed to list those.

Viraj-Sonavane committed:
Modified Pages

1.User Page added
2.NavBar added
3.Routing added but working partially
4.Browser logo added

graydosh commented on backend/api/activity/index.js:
```diff
@@ -104,4 +104,18 @@ activity.route('/:id([a-f0-9]+)')
 	}
   });
 
+
+  activity.route('/filter/:lat([a-f0-9]+)/:lon([a-f0-9]+)')
+  .get(async(req, res) => {
+	numPosts = req.body.numPosts || 20;
+	try {
+		let getobj = await Activity.find( { latitude:lat , logitude:lon ,creationDateTime: {$exists: true}} ).sort({creationDateTime : -1});
+		const statusCode = 200;
+		res.status(statusCode).send(getobj);
+	} catch(error) {
+		res.status(500).send(error);
+	}
+  });
+
+
```
Why is this route being added in this pull request?

graydosh commented on frontend/package.json:
```diff
@@ -11,9 +11,13 @@
     "prop-types": "^15.7.2",
     "react": "^16.13.1",
     "react-dom": "^16.13.1",
+    "react-feather": "^2.0.8",
+    "react-helmet": "^6.1.0",
+    "react-perfect-scrollbar": "^1.5.8",
```
Why do we need these modules? For icons we should be using the material-ui set `@material-ui/icons`. And for the most part, we shouldn't need to be manipulating the dom directly so react-helmet doesn't make sense to have. Maybe we can change the scrollbar eventually but for now it's not the highest priority

graydosh commented on frontend/public/robots.txt:
```diff
@@ -1,3 +0,0 @@
-# https://www.robotstxt.org/robotstxt.html
```
We should keep this file. Eventually we might want to use it to discourage web-scraping of our site

graydosh commented on frontend/src/Account/Acitivity/UserActivity.js:
```diff
@@ -0,0 +1,143 @@
+import React, { useState } from 'react';
+import clsx from 'clsx';
+import moment from 'moment';
+import { v4 as uuid } from 'uuid';
```
Seems 'clsx' and 'uuid' aren't in our package.json. We won't need uuid in the future so it's best not to include it at all and instead just make up a fake id to test with.

graydosh commented on frontend/src/Account/Groups/Group.js:
```diff
@@ -0,0 +1,62 @@
+} from '@material-ui/core';
+import Toolbar from './Toolbar';
+import GroupCard from './GroupCard';
+import data from './data';
```
If we're gonna pull test data from a local file, please put a comment indicating that it only for testing and should be removed in production.

graydosh commented on frontend/src/Account/Groups/Group.js:
```diff
@@ -0,0 +1,62 @@
+            container
+            spacing={3}
+          >
+            {groups.map((groups) => (
```
When using `Array.map()` on a variable with a plural name, consider making the callback variable name be singular. It's helpful for readability.

graydosh commented on frontend/src/Account/Acitivity/UserActivity.js:
```diff
@@ -0,0 +1,143 @@
+};
+
+UserActivity.propTypes = {
+  className: PropTypes.string
```
For properties that aren't required make sure to specify a default with `UserActivity.defaultProps`

graydosh commented on frontend/src/Account/Groups/GroupCard.js:
```diff
@@ -0,0 +1,126 @@
+};
+
+GroupCard.propTypes = {
+  className: PropTypes.string,
```
I mentioned in another comment, make sure to use `defaultProps` if properties aren't required. This is the last one I'll comment on so please make sure the other components also have default values specified.

graydosh commented on frontend/src/Account/NavItem.js:
```diff
@@ -0,0 +1,85 @@
+      disableGutters
+      {...rest}
+    >
+      <Button
+        activeClassName={classes.active}
+        className={classes.button}
+        component={RouterLink}
+        to={href}
+      >
```
Unfortunately it doesn't seem that changing the component type is working here. I tried running this branch of your fork and it changed the location, but the page didn't update because the router in the redux store wasn't updated properly. Maybe consider using the [`useHistory` hook](https://github.com/ReactTraining/history/blob/master/docs/api-reference.md#history.push) that the module 'history' provides as it would be easy to implement in an `onClick` callback and you wouldn't have to mess with changing the component types.

graydosh commented on frontend/src/Account/Settings/Setting.js:
```diff
@@ -0,0 +1,38 @@
+} from '@material-ui/core';
+import Notifications from './Notifications';
+import Password from './Password';
+import TopBar from '../../TopBar';
+import NavBar from '../NavBar';
```
Why are TopBar and NavBar being rendered in this component? If they are persistent across every page they should be rendered in `src/App.js`.

graydosh commented on frontend/src/Account/UpdateProfile/ProfileDetails.js:
```diff
@@ -0,0 +1,197 @@
+                label="Phone Number"
+                name="phone"
+                onChange={handleChange}
+                type="number"
```
This type should be 'tel' instead of number. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/tel

```suggestion
                type="tel"
```

graydosh commented on frontend/src/App.js:
```diff
@@ -1,11 +1,9 @@
-import Header from './Header';
-
+import TopBar from './TopBar'
 function App() {
   return (
     <>
-      <Header />
       <Main />
     </>
   );
```
I mentioned this in another comment, we should be adding elements that exist on every page (i.e. the header/navigation bar) to this component instead of to each page individually.

graydosh commented on frontend/src/Authentication/Login.js:
```diff
@@ -0,0 +1,169 @@
+                    xs={12}
+                    md={6}
+                  >
+                    <Button
+                      color="primary"
+                      fullWidth
+                      startIcon={<FacebookIcon />}
+                      size="large"
+                      variant="contained"
+                    >
+                      Login with Facebook
+                    </Button>
+                  </Grid>
+                  <Grid
+                    item
+                    xs={12}
+                    md={6}
+                  >
+                    <Button
+                      fullWidth
+                      startIcon={<GTranslate/>}
+                      size="large"
+                      variant="contained"
+                    >
+                      Login with Google
+                    </Button>
```
For now, let's remove the "Sign in with X" options from this page as we don't have that functionality available from the backend. We can add it back later if needed.

graydosh commented on frontend/src/Authentication/Register.js:
```diff
@@ -0,0 +1,150 @@
+                >
+                  Have an account?
+                  {' '}
+                  <Link component = {Link} to="/Login" >
```
If we're using the `Link` component we don't need to add the `component` property to it as it's already of that type. This also shows up on the Login page.

graydosh commented on frontend/src/Main.js:
```diff
@@ -1,13 +1,37 @@
-import React from 'react';
+import React, { Profiler } from 'react';
```
We don't need to import `Profiler` if it's not being used.

graydosh commented on frontend/src/TopBar.js:
```diff
@@ -0,0 +1,130 @@
+  const classes = useStyles();
+  return (
+    <ThemeProvider theme = {theme}> 
+    <Router>
```
We've already defined a router elsewhere in the application so defining another here could cause problems.

graydosh commented on frontend/src/TopBar.js:
```diff
@@ -0,0 +1,130 @@
+       onClose ={() => setOpen(false)}
+       >
+      <div style={{height : '100%', width : '250px'}}>
+        <MenuItem component ={Link} to="/" >
+         <IconButton size='small'>
+          Post
+         </IconButton>
+        </MenuItem>
+        <MenuItem component ={Link} to="/Activity" color='inherit' align ='right'>
```
I mentioned this in another comment, but I'm not sure specifying the `component` property will work as expected in this case. It might be more straightforward to use the history hooks to change location.
